packages/pigeon/test/java_generator_test.dart

```diff
@@ -870,6 +870,32 @@ void main() {
             'Long output = api.add((xArg == null) ? null : xArg.longValue(), (yArg == null) ? null : yArg.longValue())'));
   });
 
+  test('host Object argType', () {
+    final Root root = Root(apis: <Api>[
+      Api(name: 'Api', location: ApiLocation.host, methods: <Method>[
+        Method(
+          name: 'objectTest',
+          arguments: <NamedType>[
+            NamedType(
+                name: 'x',
+                type: const TypeDeclaration(
+                    isNullable: false, baseName: 'Object')),
+          ],
+          returnType: const TypeDeclaration.voidDeclaration(),
+        )
+      ])
+    ], classes: <Class>[], enums: <Enum>[]);
+    final StringBuffer sink = StringBuffer();
+    const JavaOptions javaOptions = JavaOptions(className: 'Api');
+    generateJava(javaOptions, root, sink);
+    final String code = sink.toString();
+    expect(code, contains('class Api'));
+    expect(code, contains('void objectTest(@NonNull Object x)'));
+    expect(
+        code, contains('ArrayList<Object> args = (ArrayList<Object>)message;'));
```

Comment:
It's probably best to omit superfluous asserts so the point of the test is more clear.